backends/gstreamer/source-element/src/source.rs

```diff
@@ -0,0 +1,174 @@
+use glib::SignalHandlerId;
+use gobject_subclass::object::*;
+use gst;
+use gst_app::{self, AppSrc, AppSrcCallbacks, AppStreamType};
+use gst_base::prelude::*;
+use gst_plugin::base_src::*;
+use gst_plugin::element::{ElementClassExt, ElementImpl};
+use gst_plugin::uri_handler::{register_uri_handler, URIHandlerImpl, URIHandlerImplStatic};
+
+macro_rules! inner_appsrc_proxy {
+    ($fn_name:ident, $arg1:ident, $arg1_type:ty, $return_type:ty) => (
+        pub fn $fn_name(&self, $arg1: $arg1_type) -> Result<$return_type, ()> {
+            match self.appsrc {
+                Some(ref appsrc) => Ok(appsrc.$fn_name($arg1)),
+                None => Err(()),
+            }
+        }
+    )
+}
+
+pub struct ServoSrc {
+    debug_category: gst::DebugCategory,
+    appsrc: Option<gst_app::AppSrc>,
+}
+
+impl ServoSrc {
+    fn new(element: &BaseSrc) -> Box<BaseSrcImpl<BaseSrc>> {
+        element.set_format(gst::Format::Time);
+        let debug_category = gst::DebugCategory::new(
+            "servosrc",
+            gst::DebugColorFlags::empty(),
+            "Servo Source",
+        );
+
+        let appsrc = gst::ElementFactory::make("appsrc", None)
+            .map(|e| e.downcast::<AppSrc>().unwrap());
+
+        Box::new(Self {
+            debug_category,
+            appsrc,
+        })
+    }
+
+    fn class_init(klass: &mut BaseSrcClass) {
+        klass.set_metadata(
+            "Servo Media Source",
+            "Source/Audio/Video",
+            "Feed player with media data",
+            "Servo developers",
+        );
+
+        let caps = gst::Caps::new_simple(
+            "video/x-raw",
+            &[
+                ("format", &"BGRA"),
+                ("pixel-aspect-ratio", &gst::Fraction::from((1, 1))),
+            ],
+        );
```

Comment:
The source element shold have ANY caps at its source pad, because we don't know what we are receiving from the network, later on, in the pipeline, the media type will be discovered.